server/core/src/main/java/org/infinispan/server/core/backup/resources/CacheResource.java

```diff
@@ -193,10 +197,7 @@ public CacheBackupEntry next() {
                }
             }
          };
-
-         int batchSize = (int) (Long.highestOneBit(ProcessorInfo.availableProcessors()) << 1);
-         log.debugf("Cache %s has file of size %d, batch size %d", cacheName, is.available(), batchSize);
-         Single<Long> restored = Flowable.fromIterable(() -> backupEntries)
+         return Flowable.fromIterable(() -> backupEntries)
                .rebatchRequests(batchSize)
                .map(entry -> {
```

Comment:
This and rest of the operators here can be moved out of this publisher and added to the one below after it is wrapped by the BlockingManager. We only want the blocking code to hit the iterator, anything else should be outside of the using if possible.